darasimiolaifa/amistad
# feat(notification): implement notification feature

## What does this PR do?
This PR implements the user notification feature

## What major activities were carried out?
- create notification component
- create action handler and reducer to handle showing notifications
- create action handler to connect notifications to the scream they are connected to
- create skeleton components to show while page is loading

## How can this be manually tested?
- Clone the repository
- Fetch the branch
- Install the dependencies
- cd into the `amistad-client` folder
- Start project by running `npm start` in the terminal
- Your browser should automatically load to show you the homepage screen below:

![home with expand scream](https://user-images.githubusercontent.com/28527393/71788869-bc0bc580-3026-11ea-885c-a7e98f70e7b7.png)
- Click the `signup` button. You should see the following page:

![signup page](https://user-images.githubusercontent.com/28527393/71532515-fd4bf900-28f3-11ea-9fdf-6a60ad58ada6.png)

- Fill in the form details as appropriate, then submit. The browser should redirect to take you to the homepage with `screams`.

![authenticated homepage with expand scream](https://user-images.githubusercontent.com/28527393/71816688-abd60380-3083-11ea-8850-96489cfa91f1.png)
- Click on the `plus` icon in the navbar and create a scream, and when it is successfully created, log out of the account by clicking on the `carriage return` icon on the `left side` of the profile card.
- On the homepage, click on `signup` and register a fresh account like you did some moments ago.
- Now, while logged in as the `second` user, either `like` or `comment` on (or do both) the scream you created with the `first` account. Then log out.
- Log in as the `first` account you created. Your notification icon should have a badge with a number indicating the number of notifications you have like this:

![authenticated page showing number of notifications](https://user-images.githubusercontent.com/28527393/72067697-b1318900-32e3-11ea-9d2a-cf4c48822dca.png)
- Click on the icon, for a moment, you should see your notifications in `red`, and after a while, they should turn `blue` like this:

![notification plane opened](https://user-images.githubusercontent.com/28527393/72067887-3157ee80-32e4-11ea-9680-22ead86f6d7c.png)
- Click on any of the notifications, and it should take you to the owner's page, and automatically open the scream dialog like this:

![Clicked notification details](https://user-images.githubusercontent.com/28527393/72068400-41240280-32e5-11ea-9e78-9d32465ae016.png)
